ChatApp/ChatApp.API/Controllers/AccountController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Security.Claims;
+using System.Threading.Tasks;
+using AutoMapper;
+using ChatApp.API.ViewModels.User;
+using ChatApp.Core.Entities;
+using ChatApp.Core.Enums;
+using ChatApp.CQRS.Commands.Users;
+using ChatApp.CQRS.Queries.Users;
+using MediatR;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Google;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace ChatApp.API.Controllers
+{
+    [Route("account")]
+    [ApiController]
+    [AllowAnonymous]
+    public class AccountController : ControllerBase
+    {
+        private readonly IMapper _mapper;
+        private readonly IMediator _mediator;
+
+        public AccountController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpPost("register")]
+        [AllowAnonymous]
+        public async Task<IActionResult> RegisterAsync([FromBody] RegisterViewModel model)
+        {
+            var user = await _mediator.Send(new RegisterUserCommand(model.FirstName, model.LastName, model.EmailAddress, model.Photo));
+
+            var result = _mapper.Map<User, UserViewModel>(user);
+
+            return Ok(result);
+        }
+
+        [HttpGet]
+        [Route("google-login")]
+        public IActionResult Login(string returnUrl)
+        {
+            return new ChallengeResult(
+                GoogleDefaults.AuthenticationScheme,
+                new AuthenticationProperties
+                {
+                    RedirectUri = Url.Action("signin-google", new { ReturnUrl = returnUrl })
+                });
+        }
+
+        [HttpGet("signin-google")]
+        public async Task<IActionResult> LoginCallback(string returnUrl = null)
+        {
```

Comment:
try this
public async Task<IActionResult> LoginCallback(string returnUrl = "~/")

Reply:
Done!